exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricConfiguration.java

```diff
@@ -282,6 +296,29 @@ public abstract Builder setMonitoredResourceDescription(
      */
     public abstract Builder setResourceAttributesFilter(Predicate<AttributeKey<?>> filter);
 
+    /**
+     * Sets the options used to configure the {@link
+     * com.google.cloud.monitoring.v3.MetricServiceClient} used to interact with the Cloud
+     * Monitoring API. This is for advanced usage and must be configured carefully.
+     *
+     * <p>Providing MetricServiceSettings will cause the exporter to ignore the values configured
+     * using:
+     *
+     * <ul>
+     *   <li>{@link MetricConfiguration.Builder#setInsecureEndpoint(boolean)}
```

Comment:
Did you explore making these functions work with or update MetricServiceSettings?

Reply:
Yes, I did try to reconcile these functions with the `setMetricServiceSettings`, but it lead to a confusing API. All of these functions eventually affect the properties of the underlying `MetricServiceSettings` object used to create the client.
 
So in a config like:
```java
    return MetricConfiguration.builder()
        .setMetricServiceSettings(metricServiceSettingsBuilder.build())
        .setCredentials(validCredentials)
        // sets the channel to use NoCredentialsProvider and plainText.
        .setInsecureEndpoint(true)
        .setMetricServiceEndpoint(someEndpoint)
        .build();
```

It would become confusing for the user to predict/understand what the final configuration holds. Ignoring these other settings when `setMetricServiceSettings` is used keeps the **implementation clean** and the intent of the API clear to the user.

We also have exposed the defaults which the users can plug-in when constructing the `MetricServiceSettings` object.